gadget/gadget.go

```diff
@@ -200,6 +200,21 @@ func (v *Volume) MinSize() quantity.Size {
 	return quantity.Size(endVol)
 }
 
+// MaxSize returns the maximum size required by a volume, as implicitly
+// defined by the size structures. It assumes sorted structures.
+func (v *Volume) MaxSize() quantity.Size {
```

Comment:
As this is identical code as `MinSize`, maybe both could share it by using an internal `Volume.size` method that accepts a function returning the size of a structure as an argument.

Reply:
Good idea. I improved it.